tools/utils.py

```diff
@@ -93,3 +94,32 @@ def write_binary(file_path, contents):
   """Write to a file opened in binary mode"""
   with open(file_path, 'wb') as fh:
     fh.write(contents)
+
+
+def delete_file(filename):
```

Comment:
`try_delete` suggests that we try, and do nothing if we fail. `delete` by itself suggests an error if we fail, but this doesn't do that. `delete_if` perhaps?

Reply:
The `try` part of `try_delete`, I think refereed to the fact the if the delete itself failed it would just ignore that fact.  

These new function never ignore failure, by design.    They do however have an early return if the file is already deleted, which I think is fine/reasonable.   I'm not sure its worth including an `if` in the name...

Reply:
I guess I could create two versions of all of these... one which expects the file to exist on entry and one that doesn't.   `delete_file` and `maybe_delete_file`?      I'm not sure its worth it though.

Reply:
@sbc100 Ah, I see what you mean. sgtm as is.